core/frontend/src/extension/Extension.ts

```diff
@@ -0,0 +1,58 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Extensions
+ */
+
+/**
+ * @beta
+ */
+export enum ActivationEvent {
+  onStartup = "onStartup",
+}
+
+/**
+ * @beta
+ */
+export type ResolveFunc = () => Promise<any>;
+
+/** Defines the format of an Extension manifest
+ * @beta
+*/
+export interface ExtensionManifest {
+  /** The extension name */
+  readonly name: string;
+  /** The extension display name */
+  readonly displayName?: string;
+  /** The extension version */
+  readonly version: string;
+  /** The extension description */
+  readonly description?: string;
+  /** The extension author */
+  readonly author: string;
+  /** The main module file to load. This should be a path to the javascript file
+   * e.g "./lib/main.js"
+   */
+  readonly main?: string;
+  /** List of activation events this Extension supports. */
+  readonly activationEvents: ActivationEvent[];
+}
+
+/** @beta */
+export interface BuildExtensionManifest extends ExtensionManifest {
+  /** Only valid when the Extension is loaded at build-time.
+   *
+   * Defines how to load the Extension manifest and
```

Comment:
and what?
If a BuildExtensionManifest is specifically about an extension loaded at build-time (I assume it is, docs don't say) then shouldn't the `module` property be required?

Reply:
it should....and we actually aren't using this interface where we should be. I'll update

Reply:
updated and used appropriately